app/models/search.rb

```diff
@@ -0,0 +1,68 @@
+class Search < ActiveRecord::Base
+  require 'date'
+  include ActiveModel::ForbiddenAttributesProtection
+
+  attr_accessible :key_words, :title, :main_type, :note_type, :min_date, :created_by,
+                  :language, :max_date
+
+  ## current advanced search
+  def advanced_search(input, params)
+    all = !params[:notes] && !params[:wikis] && !params[:maps] && !params[:comments]
+    @nodes = []
+    unless input.blank?
+      @nodes += SearchService.new.find_notes(input, 25) if params[:notes] || all
```

Comment:
Here, would it be possible (or desirable?) to eliminate the `new` method by using a method like:

```ruby
def self.find_notes(input, num)
  ss = self.new
  # ...
end
```
or something like this? Not hugely important, but I often see calls like this without a specific `new` invocation and just curious if that's a helpful simplification?